livekit-agents/livekit/agents/job.py

```diff
@@ -99,12 +107,74 @@ async def connect(
 
         await self._room.connect(self._info.url, self._info.token, options=room_options)
         self._on_connect()
+        for p in self._room.remote_participants.values():
+            self._on_participant_connected(p)
 
         _apply_auto_subscribe_opts(self._room, auto_subscribe)
 
     def shutdown(self, reason: str = "") -> None:
         self._on_shutdown(reason)
 
+    def _on_participant_connected(self, p: rtc.RemoteParticipant) -> None:
+        for filter, coro in self._participant_coro_lookup.items():
+            if filter(p):
+                task = asyncio.create_task(coro(p))
+                self._participant_tasks[p.identity] = task
+                task.add_done_callback(
+                    lambda _: self._participant_tasks.pop(p.identity)
+                )
+
+    def add_participant_task(
+        self,
+        *,
+        task_fnc: Callable[[rtc.RemoteParticipant], Coroutine[None, None, None]],
+        identity: str | None = None,
+        filter_fnc: Callable[[rtc.RemoteParticipant], bool] | None = None,
+    ):
+        """Adds a task to be run when a participant that matches the filter joins the room. In cases where
+        the participant has already joined, the task will be run immediately. The task will only be run once
+        per matching participant. In cases where a participant leaves and then rejoins with the same identity,
+        the new task will not be started until the old task has completed.
+        """
+
+        def _filter_fnc(p: rtc.RemoteParticipant) -> bool:
+            if filter_fnc and identity:
```

Comment:
move this outside of `_filter_fnc`? to ensure that we can raise on `add_participant_task` vs later

Reply:
a simpler way is to avoid this inner function.. but instead to set:

```python
if filter_fnc is None:
    if identity:
        filter_fnc = lambda p: p.identity == identity
    else:
        filter_fnc = lambda p: True
```